### ISSUE TITLE
Issue in transferring whiteboard data from group session to main session.

### ISSUE THREAD

**Issue by soheb-somani** [bigbluebutton/bigbluebutton#20690]
I found an issue in transferring whiteboard data from group session to main session. Here's how I tested it: one moderator and one viewer joined a meeting. The moderator created a group session (selected the checkbox to transfer whiteboard from the group session to the main session). The viewer joined the group session. The viewer wrote something using pen. The moderator joined the group session. The viewer made the multi-user-whiteboard option active. The moderator typed something using the whiteboard text-label thing. The moderator ended the group session. The presentation from the group session was transferred to the main session, but it didn't include everything that was written in the group session.

**Expected behavior**
Everything that was written in the whiteboard in the group session should be transferred to the main session.

**Actual behavior**
Not everything that was written in the whiteboard in the group session should be transferred to the main session. Some things were missing, in this case, whatever the meeting moderator typed.

**Screenshots**
Whiteboard in the group session
<img width="867" alt="Screenshot 2024-07-11 at 17 19 52" src="https://github.com/bigbluebutton/bigbluebutton/assets/32495387/bc5d1b64-58d8-4ce5-b434-28c00a145cb1">

Transferred PDF to the main session 
<img width="699" alt="Screenshot 2024-07-11 at 17 20 10" src="https://github.com/bigbluebutton/bigbluebutton/assets/32495387/7ab486b4-ad70-437e-8f17-e884843189c6">

**Desktop (please complete the following information):**
 - OS: MacOS 14.4.1 (tried in Windows 11 as well, same result)
 - Browser : Firefox (tried in Chrome as well, same result)
 - Version: 127.0

BigBlueButton Version : 2.7.9

**Screen-recording**

https://drive.google.com/file/d/1GaNnJ-QefJt5_X-J6PHYO0tEowXaKIBl/view?usp=share_link

**Comment by antobinary** [comment]
Thanks for reporting!

**Comment by ffdixon** [comment]
Hi @soheb-somani, do you still have the video you uploaded?  Github is only showing us the first 53 seconds.  Could you upload it to YouTube (on an unlisted link) and share the URL.

I just tried a quick test on the latest internal build of 2.7.10 and I wasn't able to reproduce the issue on the first attempt, but want to ensure I'm faithfully reproducing your example.

**Comment by soheb-somani** [comment]
Hi @ffdixon, I uploaded it on the google drive and shared the link: [https://drive.google.com/file/d/1GaNnJ-QefJt5_X-J6PHYO0tEowXaKIBl/view?usp=share_link](https://drive.google.com/file/d/1GaNnJ-QefJt5_X-J6PHYO0tEowXaKIBl/view?usp=share_link)

**Comment by antobinary** [comment]
![image](https://github.com/user-attachments/assets/9d849340-2bc6-483a-b18f-d135d6b68fef)

At 2:01 it is the viewer who circled the This is test words. That pen mark is also missing. So it looks like everthing after the empty typing box being moved or the sticky note is missing.

**Comment by antobinary** [comment]
[Screencast from 2024-07-24 06:45:44 PM.webm](https://github.com/user-attachments/assets/97b47fde-873c-4f95-8746-1c5d99c15435)
I tried to reproduce (several times) what you did without success. Any additional detail you can think of? How frequent is it? Always happening to the same moderator? Is it always cutting at the same shape? I am trying to reproduce the sequence of events around 1:23 - 1:29 but not sure how you selected the sticky note -- did you use shortcut '0'? What is the OS?

**Comment by soheb-somani** [comment]
"Any additional detail you can think of? How frequent is it?" =  **With the 2.7.9 version that we are currently using, a few different trainers (using windows) have already raised this issue and I have been able to reproduce it every time I tried (2-3 times).**

"Always happening to the same moderator?" = **No, also happens when only participants interact**

"I am trying to reproduce the sequence of events around 1:23 - 1:29 but not sure how you selected the sticky note -- did you use shortcut '0'?" - **I think sticky note is not so important. I didn't use a shortcut, I accidentally clicked on it. :)**

"What is the OS?" - I use MacOS - 14.4.1 (23E224), but we have a trainers and participants who use windows 10/11 and have had this issue.

**Comment by antobinary** [comment]
We had a call with @soheb-somani earlier today and were able to hit the issue a few times both when exporting from breakouts AND when exporting the presentation+annotation in the main room.



Looking back at the examples from @soheb-somani it seems there are several issues at play:
1) text shapes missing -- we were able to reproduce this by just adding a bunch of text annotations. 

After some investigation and comparing text annotations that were missing vs ones which displayed fine, it seems we're experiencing https://github.com/bigbluebutton/bigbluebutton/issues/19668 . However, back then the "quick fix" was to avoid crashes by ignoring text annotations which are missing the `size` property. That property missing seems to be an issue upstream in `tldraw`. Ignoring the annotations produces the situation described in this issue (#20690)

Way forward -- checking if we can fix this upstream. We need to check if BBB 3.0 (using tldraw v2) also suffers.

fine:

```
{
  "size": [
    334,
    38
  ],
  "style": {
    "isFilled": false,
    "size": "small",
    "scale": 1,
    "color": "cyan",
    "textAlign": "start",
    "font": "script",
    "dash": "draw"
  },
  "id": "4f7e0cfd-ac4c-4697-1689-2af09dc89364",
  "isModerator": true,
  "parentId": "2",
  "childIndex": 23,
  "name": "Text",
  "point": [
    -1.8873399061628016,
    365.9297489221405
  ],
  "rotation": 0,
  "text": "5. (redacted) writing on the left - middle",
  "userId": "w_aun4yhtmkvni",
  "type": "text"
}
```


not fine:

```
{
  "style": {
    "isFilled": false,
    "size": "small",
    "scale": 1,
    "color": "cyan",
    "textAlign": "start",
    "font": "script",
    "dash": "draw"
  },
  "id": "0295738f-ed65-4f45-1036-54784245f931",
  "isModerator": true,
  "parentId": "2",
  "childIndex": 24,
  "name": "Text",
  "point": [
    537.7600098275425,
    277.5240933299518
  ],
  "rotation": 0,
  "text": "6. (redacted) writing on the middle",
  "userId": "w_aun4yhtmkvni",
  "type": "text"
}
```


2) drawing (pencil) shapes missing - the example from the issue description above. I have not seen this one in the investigation yet.

3) Take a snapshot -- shapes missing -- looks like we shift everything to the left. Shapes which were [started?] too close to the left-hand side border are missing altogether. A bit of space is added to the right of the original shapes in the snapshot.

**Comment by danielpetri1** [comment]
Hi! The PDF export and breakout capture both use the same component, so for debugging purposes exporting a PDF or creating a breakout room with content saving enabled achieve the same thing. e.g., if text is missing from the breakout room capture, it won't show up in the download, too.

I believe that the pencil shapes missing are just a consequence of the text failing to render, as the PDF generation is then stopped.

**Comment by antobinary** [comment]
We have some fixes on branch `v2.7.x-release`

1) should be resolved after #20809 and #20857
2) have not been able to reproduce.
3) should be resolved now after https://github.com/bigbluebutton/bigbluebutton/pull/20831
